pkg/build/api/v1/types.go

```diff
@@ -402,6 +408,10 @@ type SourceBuildStrategy struct {
 	ForcePull bool `json:"forcePull,omitempty" description:"forces the source build to pull the image if true"`
 }
 
+// ExternalBuildStrategy defines input parameters specific to an External build.
+type ExternalBuildStrategy struct {
```

Comment:
@jimmidyson how about pointing to jenkins url (or an arbitrary url to support other systems) instead of the job url here? then annotate with the job path when the job will be created?